src/converters.ts

```diff
@@ -0,0 +1,164 @@
+   * @returns the value parsed from the string
+   */
+  auto(str: string) {
+    if (autoConverted.hasOwnProperty(str)) {
+      return autoConverted[str];
+    }
+    try {
+      return converters.number(str);
+    } catch (numErr) {
+      const strVal = converters.string(str);
+      if (strVal.charAt(0) === "'" && strVal.charAt(str.length - 1) === "'") {
+        return converters.json('"' + str.substring(1, str.length - 1) + '"');
+      } else {
+        return strVal;
+      }
+    }
+  },
+  /**
+   * Parse a number from a string
+   *
+   * @param x
+   * @returns
+   * @throws {RQLConversionError} when the string cannot be parsed
+   */
+  number(x: string): number {
+    const num = Number(x);
+    if (isNaN(num)) {
+      throw new RQLConversionError('Invalid number ' + num);
+    }
+    return num;
+  },
+  /**
+   * Parse a date from a string containing a number representing a UTC timestamp.
+   *
+   * @param x
+   * @returns
+   * @throws {RQLConversionError} when the string cannot be parsed
+   */
+  epoch(x: string): Date {
+    const date = new Date(converters.number(x));
+    if (isNaN(date.getTime())) {
+      throw new RQLConversionError(`Invalid date ${x}`);
+    }
+    return date;
+  },
+  /**
+   * Parse a date from a string in ISO date format
+   *
+   * @param x
+   * @returns
+   */
+  isodate(x: string): Date {
+    // four-digit year
+    let date = '0000'.substr(0, 4 - x.length) + x;
+    // pattern for partial dates
+    date += '0000-01-01T00:00:00Z'.substring(date.length);
+    return converters.date(date);
+  },
+  /**
+   * Parse a date from a string in YYYY-MM-DDTHH:mm:ss.SSSZ format
+   *
+   * Milliseconds and "Z" are optional.
+   *
+   * @param x
+   * @returns
+   * @throws {RQLConversionError} when the string cannot be parsed
+   */
+  date(x: string): Date {
+    // tslint:disable-next-line: tsr-detect-unsafe-regexp
+    const isoDate = /^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})(?:\.(\d{1,3}))?Z$/.exec(x);
+    let date;
+    if (isoDate) {
+      date = new Date(
+        Date.UTC(+isoDate[1], +isoDate[2] - 1, +isoDate[3], +isoDate[4], +isoDate[5], +isoDate[6], +isoDate[7] || 0)
+      );
+    } else {
+      date = new Date(x);
+    }
+    if (isNaN(date.getTime())) {
+      throw new RQLConversionError(`Invalid date ${x}`);
+    }
+    return date;
+  },
+  /**
+   * Parse a boolean from a string. The string must be 'true' (case insensitive), otherwise it is false.
+   *
+   * @param x
+   * @returns
+   */
+  boolean(x: string): boolean {
+    return x.toLowerCase() === 'true';
+  },
+  /**
+   * Decode a URL encoded string
+   *
+   * @param str
+   * @returns
+   */
+  string(str: string): string {
+    return decodeURIComponent(str);
+  },
+  /**
+   * Parse a case-insensitive regular expression from a string.
+   *
+   * @param x
+   * @returns
+   * @throws {RQLConversionError} when the string cannot be parsed
+   */
+  re(x: string): RegExp {
```

Comment:
hmm. I agree with you but it's a little tricky since it's intended to match RQL syntax here. We could put in a mapping function from RQL syntax -> converter function name. Not sure.

Reply:
this doesn't bother me as much, if you really want to make a mapping from name to function then go for it

Reply:
interesting, none of the specs actually address the "re:" / "RE:" types. 

RQL suggests the following types to be supported:

string - Indicates the value string should not be coerced, it should remain a string.
number - Indicates the value string should be coerced to a number.
boolean - Indicates the value string should be coerced to a boolean. A value of "true" should indicate true and a value of "false" should indicate false.
epoch - Indicates the value string should be treated as the milliseconds since the epoch and coerced to a date.

http://dundalek.com/rql/draft-zyp-rql-00.html

Reply:
This would be more than a breaking code change though, this would break the compatibility with RQL usage.

Reply:
Then don't change it. There is something to be gained to have brevity in URL arguments

Reply:
oh, didn't know it was dictated by the rql syntax. I _guess_ we can leave it :cry:

Reply:
changed